### PR TITLE
Test improvements and new tests to verify overlayfs on ID-mapping support.

### PR DESCRIPTION
**Pull request by ctalledo**

### CODE REVIEW

File: tests/dind/build.bats [hunk 1]
```diff
@@ -212,6 +213,10 @@ function teardown() {
 
 @test "commit with removed inner image" {
 
+  if [[ $(get_platform) == "arm64" ]]; then
+	  skip "syscont-inner-img not supported on arm64."
```
Comment (rodnymolina) [inline tests/dind/build.bats:217]: I uploaded an arm `syscont-inner-img` in the past, probably it got overwritten with an amd64-only one.

**Review by rodnymolina — APPROVED** [review]
Minor questions/comments only.

LG

File: tests/dind/build.bats [hunk 2]
```diff
@@ -36,7 +36,7 @@ function teardown() {
   # do a docker build with appropriate dockerfile
   pushd .
   cd tests/dind
-  docker build --no-cache -t sc-with-inner-img:latest .
+  DOCKER_BUILDKIT=0 docker build --no-cache -t sc-with-inner-img:latest .
```
[the remark this answers is not in the document]
Comment (ctalledo) [reply tests/dind/build.bats:40]: It's not the default anymore, buildkit is; and unfortunately buildkit does not honor the "default-runtime" config, so it always uses runc. This is why we have to explicitly switch to the traditional builder.

File: tests/sysmgr/mount.bats [hunk 3]
```diff
@@ -10,6 +10,100 @@ load ../helpers/docker
...
+	elif sysbox_using_shiftfs_on_overlayfs; then
+		docker exec "$syscont" sh -c "findmnt | egrep \"^/\" | grep shiftfs"
+		[ "$status" -eq 0 ]
+	else
+		docker exec "$syscont" sh -c "findmnt | egrep \"^/\" | grep \"/var/lib/sysbox\""
+		[ "$status" -eq 0 ]
+	fi
+
+	# Verify implicit bind mounts use idmapping or shiftfs or chown
+   local kernel_rel=$(uname -r)
```
[the remark this answers is not in the document]
Comment (ctalledo) [reply tests/sysmgr/mount.bats]: Done.

File: tests/helpers/k3s.bash [hunk 4]
```diff
@@ -115,8 +115,16 @@ function k3s_cluster_setup() {
   [ "$status" -eq 0 ]
   kubectl config set-context default
   [ "$status" -eq 0 ]
-  docker cp ${master_node}:/etc/rancher/k3s/k3s.yaml kubeconfig
+
+  # TODO: for some reason "docker cp" is failing to copy to/from subdirs in the
```
[the remark this answers is not in the document]
Comment (ctalledo) [reply tests/helpers/k3s.bash:119]: That command will work for sure. The failure is specific to `docker cp`.

File: tests/dind/build.bats (hunk 1 above)
Comment (ctalledo) [reply tests/dind/build.bats:217]: Oh I see ... makes sense; not a big deal though.

**Pull request merged**